spec/unit/iiif/v3/presentation/sequence_spec.rb

```diff
@@ -1,98 +1,176 @@
-      'label' => 'Book 1',
-      'description' => 'A longer description of this example book. It should give some real information.',
-      'thumbnail' => {
-        'id' => 'http://www.example.org/images/book1-page1/full/80,100/0/default.jpg',
-        'service'=> {
-          '@context' => 'http://iiif.io/api/image/2/context.json',
-          'id' => 'http://www.example.org/images/book1-page1',
-          'profile' => 'http://iiif.io/api/image/2/level1.json'
-        }
-      },
-      'attribution' => 'Provided by Example Organization',
-      'rights' => [{'id' => 'http://www.example.org/license.html'}],
-      'logo' => 'http://www.example.org/logos/institution1.jpg',
-      'see_also' => 'http://www.example.org/library/catalog/book1.xml',
-      'service' => {
-        '@context' => 'http://example.org/ns/jsonld/context.json',
-        'id' =>  'http://example.org/service/example',
-        'profile' => 'http://example.org/docs/example-service.html'
-      },
-      'related' => {
-        'id' => 'http://www.example.org/videos/video-book1.mpg',
-        'format' => 'video/mpeg'
-      },
-      'within' => 'http://www.example.org/collections/books/',
-      # Sequence
-      'metadata' => [{'label'=>'Author', 'value'=>'Anne Author'}],
-      'items' => [{
-        'id' => 'http://www.example.org/iiif/book1/canvas/p1',
-        'type' => 'Canvas',
-        'label' => 'p. 1',
-        'height' => 1000,
-        'width' => 750,
-        'content'=>  []
-      }],
-      'viewing_hint' => 'paged',
-      'start_canvas' => 'http://www.example.org/iiif/book1/canvas/p2',
-      'viewing_direction' => 'right-to-left',
-    }
+  describe '#array_only_keys' do
+    it 'items' do
+      expect(subject.array_only_keys).to include('items')
+    end
+  end
+
+  describe '#legal_viewing_hint_values' do
+    it 'contains the expected values' do
+      expect(subject.legal_viewing_hint_values).to contain_exactly('individuals', 'paged', 'continuous', 'auto-advance')
+    end
   end
 
   describe '#initialize' do
     it 'sets type to Sequence by default' do
       expect(subject['type']).to eq 'Sequence'
     end
     it 'allows subclasses to override type' do
-      sub = subclass_subject.new
+      subclass = Class.new(described_class) do
+        def initialize(hsh={})
+          hsh = { 'type' => 'a:SubClass' }
+          super(hsh)
+        end
+      end
+      sub = subclass.new
       expect(sub['type']).to eq 'a:SubClass'
     end
   end
 
-  describe "#{described_class}.define_methods_for_array_only_keys" do
-    it_behaves_like 'it has the appropriate methods for array-only keys v3'
-  end
-
-  describe "#{described_class}.define_methods_for_string_only_keys" do
-    it_behaves_like 'it has the appropriate methods for string-only keys v3'
-  end
-
-  describe "#{described_class}.define_methods_for_any_type_keys" do
-    it_behaves_like 'it has the appropriate methods for any-type keys v3'
-  end
-
   describe '#validate' do
```

Comment:
The `Sequence` class declares the allowed values for `viewingHint` and `viewingDirection` but the `AbstractResource` class validation does the actual checking:

https://github.com/sul-dlss/osullivan/blob/development/lib/iiif/v3/abstract_resource.rb#L141-L157

and so the tests of that checking are associated:  https://github.com/sul-dlss/osullivan/blob/development/spec/unit/iiif/v3/abstract_resource_spec.rb#L67-L90